elasticsearch/configure.sls

```diff
@@ -0,0 +1,108 @@
+{% from "elasticsearch/map.jinja" import elasticsearch with context %}
+
+include:
+  - .service
+  - .repository
+
+# Create folder and set permissions
+{% set conf_dir = elasticsearch.conf_folder %}
+{% set log_dir = elasticsearch.log_folder %}
+
+{% for dir in (conf_dir, log_dir) %}
+create_directory{{ dir|replace('/', '_') }}to_ensure_proper_permissions:
+  file.directory:
+    - name: {{ dir }}
+    - user: elasticsearch
+    - group: elasticsearch
+    - mode: 0700
+    - makedirs: True
+{% endfor %}
+
+# Update the I/O scheduler if using SSD to improve write throughput https://www.elastic.co/guide/en/elasticsearch/guide/current/hardware.html
+{% for device_name in salt.grains.get('SSDs') %}
+update_io_scheduler_for_{{device_name}}:
+  cmd.run:
+    - name: echo noop | tee /sys/block/{{ device_name }}/queue/scheduler
+{% endfor %}
+
+# Update Heap Size according to available RAM https://www.elastic.co/guide/en/elasticsearch/reference/current/heap-size.html
+{% set heap_max = salt.grains.get('mem_total', 0) // 2 %}
+{% set heap_size = heap_max if heap_max < 31744 else 31744 %}
+{% if elasticsearch.elastic_stack %}
+config_jvm_options:
+  file.replace:
+  - name: '{{ elasticsearch.conf_folder }}/jvm.options'
+  - pattern: '^-Xm[sx]\d\w'
+  - repl: '#-Xm[sx]\d\w'
```

Comment:
This looks like it will replace the value with the contents of the regex rather than with any capture groups so that it will not only comment out the line, but eliminate the original string.